src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -104,13 +109,43 @@ protected override void ExecuteCore()
                 PlatformLibraryName,
                 IsSelfContained);
 
+            Dictionary<PackageIdentity, StringBuilder> packagesThatWhereFiltered = null;
```

Comment:
This chunk of logic should be encapsulated in a method to enable easier reading.